include/dma.h

```diff
@@ -270,6 +273,33 @@ static inline int z_impl_dma_stop(struct device *dev, u32_t channel)
 	return api->stop(dev, channel);
 }
 
+/**
+ * @brief get length of pending bytes in DMA buffer.
+ *
+ * Implementations must check the validity of the channel ID passed in and
+ * return -EINVAL if it is invalid.
+ *
+ * @param dev     Pointer to the device structure for the driver instance.
+ * @param channel Numeric identification of the channel where the transfer was
+ *                being processed
+ *
+ * @retval non-negative if successful.
+ * @retval Negative errno code if failure.
+ */
+__syscall int dma_pending_size(struct device *dev, u32_t channel);
+
+static inline int z_impl_dma_pending_size(struct device *dev, u32_t channel)
+{
+	const struct dma_driver_api *api =
+		(const struct dma_driver_api *)dev->driver_api;
+
+	if (api->pending_size) {
```

Comment:
Is this check really needed?
I saw that all other APIs are not doing this check. Maybe this is somehow enforced by the compiler.
Have you tried to call pending_size when NULL just to double check?

Reply:
Changed `pending_size` to `pending_len`. Thanks for the suggestion. 

The NULL check is necessary here unless every DMA driver implements this function. The reason others don't do this is because all DMA drivers have implemented the all API functions except this new one. 

To remove this check, I have to add en empty function to all of other DMA drivers.

Reply:
> Changed `pending_size` to `pending_len`. Thanks for the suggestion.
> 
> The NULL check is necessary here unless every DMA driver implements this function. The reason others don't do this is because all DMA drivers have implemented all other API functions except this new one.
> 
> To remove this check, I have to add en empty function to all of other DMA drivers.

OK.
Maybe in a different commit someone should better add same check in other apis as well.
start and stop should be always there, but I guess that reload may be not present (I added it for stm32 only recently).

Reply:
your doubt is reasonable. I can see only the following three DMA drivers have implemented `reload` so far though there are six DMA drivers:

```
$ grep -rn reload * 
dma_cavs.c:358:	.reload = dw_dma_reload,
dma_sam0.c:418:	.reload = dma_sam0_reload,
dma_stm32f4x.c:566:	.reload		 = dma_stm32_reload,
```

Reply:
So, for those drivers calling reload should result in a crash.